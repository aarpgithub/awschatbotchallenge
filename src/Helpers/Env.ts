export class Env {
  static getString(key: string): string {
    return process.env[key] || "";
  }

  static getInt(key: string): Number {
    return parseInt(process.env[key] || "0");
  }
}
