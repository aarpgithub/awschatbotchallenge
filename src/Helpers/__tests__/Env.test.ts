import { Env } from "../Env";

it("Env.getInt() should expose number properties as numbers or null", () => {
  const testcases = ["one", "two", "three"];
  const samples = testcases.map(e => {
    return Env.getInt(e);
  });
  expect(samples).toMatchSnapshot();
});

it("Env.getString() should expose string properties as strings of null", () => {
  const testcases = ["one", "two", "three"];
  const samples = testcases.map(e => {
    return Env.getInt(e);
  });
  expect(samples).toMatchSnapshot();
});
