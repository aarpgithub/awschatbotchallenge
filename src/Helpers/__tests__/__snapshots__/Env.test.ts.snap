// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Env.getInt() should expose number properties as numbers or null 1`] = `
Array [
  0,
  0,
  0,
]
`;

exports[`Env.getString() should expose string properties as strings of null 1`] = `
Array [
  0,
  0,
  0,
]
`;
